go/types/set_iterator.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package types
+
+import "github.com/attic-labs/noms/go/d"
+
+type SetIterator interface {
```

Comment:
Abstractly yes, but practically no:

A Next() function makes sense to return a Value, but only for Sets and Lists; Maps would return (Value, Value).

And AdvanceTo isn't possible on Lists.

So this Iterator could only be used (in our current data structures) on a Set.